### ISSUE TITLE
Implement Training Mode with Specific Scenario Practice

### ISSUE THREAD

**Issue by jkraybill** [jkraybill/poker-game-manager#13]
### Vision
Create a training system that allows players to practice specific poker scenarios repeatedly until mastered. This is essential for players who want to improve specific aspects of their game - whether it's bubble play, short stack strategy, or multi-way pots.

### Key Features

1. **Scenario Library**
   - Pre-built scenarios for common poker situations
   - Bubble play with various stack sizes
   - Heads-up practice (deep and short)
   - Multi-way pot dynamics
   - ICM pressure situations
   - Short stack play (10-20bb)
   - Deep stack play (200bb+)

2. **Real-time Coaching**
   - Provide advice before decisions
   - Explain optimal play reasoning
   - Show EV calculations
   - Highlight common mistakes

3. **Progress Tracking**
   - Track performance in each scenario type
   - Show improvement over time
   - Identify weak areas needing more practice
   - Achievement system for mastery

### API Example
```javascript
// Practice specific scenarios
const trainer = new ScenarioTrainer();
trainer.loadScenario('bubble-play-10bb');
trainer.loadScenario('heads-up-deepstack');
trainer.loadScenario('multiway-pot-position');

// Get real-time coaching
trainer.on('decision:required', (state) => {
  const advice = trainer.getAdvice(state);
  // Returns: { recommended: Action, reasoning: string, ev: number }
});

// Track progress
trainer.on('scenario:completed', (result) => {
  console.log(`Score: ${result.score}/100`);
  console.log(`Mistakes: ${result.mistakes}`);
  console.log(`Key learning: ${result.lesson}`);
});
```

### Success Criteria
- [ ] 20+ pre-built training scenarios
- [ ] Customizable scenario creation
- [ ] Real-time optimal play calculation
- [ ] Progress tracking per scenario
- [ ] Performance: Advice generation < 50ms
- [ ] Export training history

### Priority
**Priority 2** - Championship Features (Next Month)

This directly supports players who want to improve specific aspects of their game through deliberate practice.
